workflow-service/src/test/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImplTest.java

```diff
@@ -394,6 +401,101 @@ void testUpdateWorkFlowTask() {
+	void testUpdateWorkFlowCheckerTaskStatusWithValid() {
+		// given
+		// master workflow execution
+		UUID workFlowExecutionId = UUID.randomUUID();
+		// workflow checker
+		String workFlowCheckerTaskName = "testWorkFlowTask";
+		String workFlowCheckerName = "testWorkFlowCheckerName";
+		UUID workFlowCheckerDefinitionId = UUID.randomUUID();
+		UUID projectId = UUID.randomUUID();
+
+		// when
+		// master workflow execution
+		WorkFlowExecution masterWorkFlowExecution = WorkFlowExecution.builder().status(WorkFlowStatus.FAILED)
+				.projectId(projectId).workFlowDefinitionId(UUID.randomUUID()).build();
+		masterWorkFlowExecution.setId(UUID.randomUUID());
+		Mockito.when(this.workFlowRepository.findById(Mockito.any())).thenReturn(Optional.of(masterWorkFlowExecution));
+
+		// workflow checker definition
+		WorkFlowDefinition workFlowDefinition = WorkFlowDefinition.builder().name(workFlowCheckerName).build();
+		workFlowDefinition.setId(workFlowCheckerDefinitionId);
+		// workflow checker task definition
+		WorkFlowTaskDefinition workFlowCheckerTaskDefinition = WorkFlowTaskDefinition.builder()
+				.workFlowDefinition(workFlowDefinition).name(workFlowCheckerTaskName).build();
+		workFlowCheckerTaskDefinition.setId(UUID.randomUUID());
+		Mockito.when(this.workFlowTaskDefinitionRepository.findFirstByNameAndWorkFlowDefinitionType(Mockito.any(),
+				Mockito.any())).thenReturn(workFlowCheckerTaskDefinition);
+
+		// workflow checker task execution
+		WorkFlowExecution workFlowCheckerExecution = WorkFlowExecution.builder().status(WorkFlowStatus.IN_PROGRESS)
+				.projectId(projectId).workFlowDefinitionId(workFlowCheckerDefinitionId).build();
+		workFlowCheckerExecution.setId(UUID.randomUUID());
+		Mockito.when(this.workFlowRepository.findByMasterWorkFlowExecution(Mockito.any()))
+				.thenReturn(List.of(workFlowCheckerExecution));
+
+		WorkFlowTaskExecution workFlowTaskExecution = WorkFlowTaskExecution.builder().arguments("test").results("res")
+				.status(WorkFlowTaskStatus.FAILED).workFlowTaskDefinitionId(workFlowCheckerTaskDefinition.getId())
+				.workFlowExecutionId(workFlowCheckerExecution.getId()).build();
+		workFlowTaskExecution.setId(UUID.randomUUID());
+		Mockito.when(this.workFlowTaskRepository.findByWorkFlowExecutionIdAndWorkFlowTaskDefinitionId(Mockito.any(),
+				Mockito.any())).thenReturn(List.of(workFlowTaskExecution));
+
+		this.workFlowService.updateWorkFlowCheckerTaskStatus(workFlowExecutionId, workFlowCheckerTaskName,
+				WorkFlowTaskStatus.COMPLETED);
+
+		// then
+		ArgumentCaptor<WorkFlowTaskExecution> argument = ArgumentCaptor.forClass(WorkFlowTaskExecution.class);
+		Mockito.verify(this.workFlowTaskRepository, Mockito.times(1)).save(argument.capture());
+		assertEquals(argument.getValue().getWorkFlowTaskDefinitionId(), workFlowCheckerTaskDefinition.getId());
+		assertEquals(argument.getValue().getWorkFlowExecutionId(), workFlowCheckerExecution.getId());
+		assertEquals(argument.getValue().getStatus().toString(), "COMPLETED");
```

Comment:
This can be simplified because there is no need to use `toString`: use `WorkStatus.COMPLETE` instead of `"COMPLETE"` string

Reply:
Resolved.